distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Write_Extensions.enso

```diff
@@ -40,8 +42,24 @@ polyglot java import org.enso.base.Array_Builder
 Text.write : (File|Text) -> Encoding -> Existing_File_Behavior -> Problem_Behavior -> File ! Encoding_Error | Illegal_Argument | File_Error
 Text.write self path encoding=Encoding.utf_8 on_existing_file=Existing_File_Behavior.Backup on_problems=Problem_Behavior.Report_Warning =
     bytes = self.bytes encoding on_problems
-    file = File.new path
-    r = on_existing_file.write file stream->
+
+    actual = File.new path
+
+    is_enabled = Context.Output.is_enabled
+
+    if_existing = if is_enabled then on_existing_file else
```

Comment:
I wonder if this logic might be moved into Existing_File_Behavior.

Reply:
yes thats exactly what I expect to do as we build it out.

Reply:
I agree that it would be good to move this to some common place. But OK with doing that later, once we actually have >1 place using it.